gateway/src/apicast/executor.lua

```diff
@@ -47,8 +69,8 @@ local function shared_build_context(executor)
 
     if not context then
         context = build_context(executor)
-
         ctx.context = context
+        store_original_request(ctx)
```

Comment:
Does this guarantee that the original request will be initialized on every request?

In other words, `store_original_request` only runs when the context has not been initialized, but `store_original_request` fails in some phases, so I wonder if the following case is possible:
- The body of this `if` runs and initializes `ctx.context`
- `store_original_request` is called but fails to assign the original request data
- The body of this `if` is not executed again and the original request data is never initialized

Reply:
due to error happens in the same function call, I do not think that it's the case.

Reply:
@eloycoto I've reviewed the latest changes of the PR and everything looks good to me except this. Maybe I'm missing something.

I see 2 scenarios:
- `context` is nil. `store_original_request` is called and it fails (there's a `pcall` so there are cases where it can fail). In that case, the original request would never be initialized in the context.
- `context` is nil, but when `store_original_request is called it never fails. In that case, the `pcall` would be unnecessary.

What am I missing?